### DISCUSSION TITLE
MenuItem DataTemplate creating duplicate MenuItem

### DISCUSSION THREAD

**Discussion by stephencanzanese** [AvaloniaUI/Avalonia#11571] [General]
I am currently trying to set the DataTemplate of the MenuItems to a ToggleButton. This works fine until I try to set a binding to the Content property of the ToggleButton. When I bind to the Content property, a second ToggleButton is created inside the first toggle button. If I set Content to something static like "Test", it causes a StackOverflow. I assume this is beacuse it is continualy creating nested ToggleButtons. 

![CntrlDmn](https://github.com/AvaloniaUI/Avalonia/assets/26125073/5deb73a0-8236-4998-9e57-dbdfc52c5202)

AXAML:
`<Menu Background="White" >
			<MenuItem Name="CntrlDomain" Header="Control Domain" Items="{Binding CDList}">
				<MenuItem.DataTemplates>
					<DataTemplate>
						<ToggleButton Content="{Binding Name}" Height="25" Width="100"/>
					</DataTemplate>
				</MenuItem.DataTemplates>
			</MenuItem>
		</Menu>`
VM:
`public class MainWindowVM
    {
        public ObservableCollection<TestClass> CDList { get; set; }

        public MainWindowVM()
        {
            CDList = new();
            CDList.Add(new("test1", false));
            CDList.Add(new("test2", true));
            CDList.Add(new("test3", false));
            CDList.Add(new("test4", true));
        }
    }

    public class TestClass
    {
        public string Name { get; set; }

        public bool Selected { get; set; }

        public TestClass(string name, bool selected)
        {
            Name = name;
            Selected = selected;
        }
    }	`
If I bind IsChecked to Selected, It works fine with no issues. the issue only arises when I bind to Name.

**Comment by maxkatz6** [comment]
DataTemplate without DataType inside of DataTemplates collection is an unsupported scenario. In 11.0 it will throw an error as well. You have a recursion, ad DataTemplates is inherited.

**Reply by stephencanzanese** [reply]
Bless your heart! That's it! Thank you, I've been banging my head against the wall for hours.
